server.js

```diff
@@ -1,33 +1,120 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import endpoints from 'express-list-endpoints'
+
+import booksData from './data/books.json'
+import { pagination, filterOnLanguage, filterOtherLanguages, sortByNumPages } from './bookqueries'
+import { addBook, validateBookInput } from './postBook'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
+const fs = require('file-system')
+
+const ERROR_MSG = {
+  status_code: 404,
+  error: 'no books found!'
+}
+
+const INPUT_ERROR = {
+  status_code: 400,
+  error: 'invalid input!'
+}
+
+const ERROR_PAGE = {
+  status_code: 400,
+  error: 'Please define a page',
+  example: 'books?page=1'
+}
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+const bookDefinition = [
+  {
+    fieldName: "bookID",
+    fieldType: "number"
+  },
+  {
+    fieldName: "title",
+    fieldType: "string"
+  },
+  {
+    fieldName: "authors",
+    fieldType: "string"
+  },
+  {
+    fieldName: "average_rating",
+    fieldType: "number"
+  },
+  {
+    fieldName: "isbn",
+    fieldType: "number"
+  },
+  {
+    fieldName: "isbn13",
+    fieldType: "number"
+  },
+  {
+    fieldName: "language_code",
+    fieldType: "string"
+  },
+  {
+    fieldName: "num_pages",
+    fieldType: "number"
+  },
+  {
+    fieldName: "ratings_count",
+    fieldType: "number"
+  },
+  {
+    fieldName: "text_reviews_count",
+    fieldType: "number"
+  }
+]
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(endpoints(app))
+})
+
+app.get('/books', (req, res) => {
+  let books = booksData
+  const page = req.query.page || 0
+  const language = req.query.lang
+  const otherlanguages = req.query.otherlang
+  const sorted = req.query.sort
+  
+  if (language) {
+    books = filterOnLanguage(books, language, ERROR_MSG, res)
+  } else if (otherlanguages){
+    books = filterOtherLanguages(books, language, ERROR_MSG, res)
+  }
+
+  if(sorted) {
+    books = sortByNumPages(books, sorted)
+  }
+  
+  const currentPage = pagination(books, page)
+  res.json(currentPage)
+})
+
+app.post('/books', (req, res) => {
+  const input = req.body
```

Comment:
If you specified the input you will get, you could decide for yourself which fields should come from the user and which ones should be generated by you. [Here](https://github.com/Technigo/project-express-api/pull/90/files)'s an example from one of your classmates :)